perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -1567,7 +1567,15 @@ public void run() {
 				exceptionMessage = "Cannot synchronize group ";
 				log.error(exceptionMessage + group, e);
 				exceptionMessage+= "due to unexpected exception: " + e.getClass().getName() + " => " + e.getMessage();
-				throw e;
+				try {
+					throw e;
+				} catch (PrivilegeException pe) {
+					//FIXME Don't know why, but this catch block for PrivilegeException must be here
+					failedDueToException = true;
+					exceptionMessage = "Cannot synchronize group ";
+					log.error(exceptionMessage + group, pe);
+					exceptionMessage+= "due to unexpected exception: " + pe.getClass().getName() + " => " + pe.getMessage();
+				}
 			} finally {
```

Comment:
I have no idea why the compiler forced me to add the "throw e" into a try block with PrivilegeException catching. It is catching the general Exception just few lines before. Exception extends PerunException, which extends PrivilegeException. No idea why the catching of the general Exception doesn't work for the PrivilegeException.